### ISSUE TITLE
How to use this system ?

### ISSUE THREAD

**Issue by wahbi** [neicnordic/LocalEGA#13]
### Description
after building all containers via docker-compose up -d , the user does not know how to use the system , the documentation in this link https://localega.readthedocs.io/en/latest/ is not clear enough .

the questions are :
how to add a user from cega ? does cega container have a web interface ?
how to upload a file ? 
how to download a file ? 

### Proposed solution
the answers here will help me in implementing a local ega in tunisia , thank you very much

**Comment by wahbi** [comment]
just one detail , our localEGA in tunisia is not meant to connect with the EGA in espagne , it's gonna be isolated .

**Comment by viklund** [comment]
Sorry this is taking so long. We have been a bit busy lately. We will get back to this within the next few weeks.

**Comment by viklund** [comment]
Ok, this answer should give you some pointers as to how to proceed.

We are using a python script to boostrap all the configuration needed for the system, you can find it in https://github.com/neicnordic/LocalEGA-deploy-init

How to add a user from cega? Does cega container have a web interface?
======================================================================

There is no user management within localega all of that is outsourced to other
services. Currently the easiest way is to use the users microservice in the
tests directory:

/tests/_common/users.py

That wants a json-file as third argument with users.

The json should be an array of objects with keys for username, uid,
passwordHash, gecos, sshPublicKey and enabled.

Running that service and then configuring localega to use that endpoint for
the users. Note that this is not a _prodcution_ service in its current state.


How to upload a file?
=====================

Once the above is in place, the inbox service needs to be configured. We are
using the Minna inbox (https://github.com/neicnordic/LocalEGA-inbox) and that
can use the above service for user management and then you should have an
SFTP-based inbox on the ip of that address.


How to download a file?
=======================

Currently we're using a different set of microservices that take care of this
called the ega-data-api, they are implemented in Java and can be found here:

https://github.com/neicnordic/ega-data-api

In short what it does is

1. decrypt the file using the header from DB and encrypted part from
   Archive/Vault
2. have a way to get the private key for decryption in a secure manner
3. provide an endpoint (https preferrably) where the user authenticates in
   order to get permissions to a file

**Comment by blankdots** [comment]
@wahbi we will consider this closed, but feel free to open another issue if you have further questions.
